ConcernsCaseWork/ConcernsCaseWork.API/UseCases/CaseActions/Decisions/DeleteDecision.cs

```diff
@@ -0,0 +1,69 @@
+﻿using ConcernsCaseWork.API.Contracts.ResponseModels.Concerns.Decisions;
+using ConcernsCaseWork.API.Factories.Concerns.Decisions;
+using ConcernsCaseWork.Data.Gateways;
+using System.ComponentModel.DataAnnotations;
+
+namespace ConcernsCaseWork.API.UseCases.CaseActions.Decisions
+{
+	public class DeleteDecision : IUseCaseAsync<DeleteDecisionRequest, DeleteDecisionResponse>
+	{
+		private readonly IConcernsCaseGateway _concernsCaseGateway;
+		private readonly IDecisionFactory _decisionFactory;
+		private readonly ICreateDecisionResponseFactory _createDecisionResponseFactory;
+
+		public DeleteDecision(IConcernsCaseGateway concernsCaseGateway, IDecisionFactory decisionFactory, ICreateDecisionResponseFactory createDecisionResponseFactory)
+		{
+			_concernsCaseGateway = concernsCaseGateway ?? throw new ArgumentNullException(nameof(concernsCaseGateway));
```

Comment:
SUGGESTION:

For future ref, we don't need the argument null exception, because its the depedency injection container handling this and if it can't resolve it would throw an exception anyway

Reply:
I pretty much copied the existing approaches but good to know we aren't adding this going forward. Will resolve within this PR.

Reply:
I took these out however the tests in ConcernsCaseWork.API.Tests.UseCases.CaseActions.Decisions.AllDecisionUseCasesTests.All_Constructors_Guard_Against_Null failed as they are looking for guard clauses.

Therefore decided to leave them in for consistency.